src/components/Layout/Header/NavBar/UserMenu.tsx

```diff
@@ -89,14 +92,44 @@ type WalletButtonProps = {
   onConnect: () => void
 } & Pick<InitialState, 'walletInfo'>
 
-const WalletButton = ({ isConnected, walletInfo, onConnect }: WalletButtonProps) => {
+const WalletButton: FC<WalletButtonProps> = ({ isConnected, walletInfo, onConnect }) => {
+  const [walletLabel, setWalletLabel] = useState('')
+  const bgColor = useColorModeValue('gray.300', 'gray.800')
+
+  useEffect(() => {
+    if (!walletInfo || !walletInfo.meta) return setWalletLabel('')
+    if (walletInfo.meta.address) {
+      ensReverseLookup(walletInfo.meta.address).then(ens => {
+        if (!ens.error) return setWalletLabel(ens.name)
+        // added this check here again to satisfy typescript
+        if (walletInfo.meta && walletInfo.meta.address) setWalletLabel(walletInfo.meta.address)
```

Comment:
```suggestion
        setWalletLabel(walletInfo?.meta?.address ?? '')
```

this works the same a lil more concise

Reply:
ok using this then

Reply:
@amitojsingh366 it's worth checking out the docs on [Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) for more info on this.

Optional chaining is generally preferred over explicitly testing and short-circuiting.